File: ios/Classes/FlutterNative3dPlugin.swift
import Flutter
import UIKit

public class FlutterNative3dPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let factory = Native3DPlatformViewFactory(messenger: registrar.messenger())
        registrar.register(factory, withId: "flutter_native_3d/native3d_view")
    }
}

File: android/src/main/kotlin/com/flutternative3d/flutter_native_3d/FlutterNative3dPlugin.kt
package com.flutternative3d.flutter_native_3d

import io.flutter.embedding.engine.plugins.FlutterPlugin

class FlutterNative3dPlugin : FlutterPlugin {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        binding.platformViewRegistry.registerViewFactory(
            "flutter_native_3d/native3d_view",
            Native3DPlatformViewFactory(binding.binaryMessenger)
        )
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}
